lib/Document.php

```diff
@@ -245,13 +245,14 @@ public function downloadStream($output = null): StreamInterface {
      * Uploads data to the document specified in the constructor.
      *
      * @param string|resource|StreamInterface|callable|\Iterator $feedData The contents of the feed to be uploaded
+     * @param ?string $charset An optional charset for the document to upload
      *
      * @return void
      */
-    public function upload($feedData): void {
+    public function upload($feedData, ?string $charset = null): void {
         $response = $this->client->put($this->url, [
             RequestOptions::HEADERS => [
-                "content-type" => $this->contentType,
+                "content-type" => self::get_content_type_with_charset($this->contentType, $charset),
                 "host" => parse_url($this->url, PHP_URL_HOST),
             ],
```

Comment:
if we just give the new parameter a default of `'utf-8'`, we can remove a bunch of code, because then the 'content-type' header can become `"$contentType; charset=$charset"`, and the `get_content_type_with_charset` method can be removed altogether :)